src/wp-includes/link-template.php

```diff
@@ -1469,10 +1469,15 @@ function get_edit_post_link( $post = 0, $context = 'display' ) {
 		return;
 	}
 
-	if ( $post_type_object->_edit_link ) {
-		$link = admin_url( sprintf( $post_type_object->_edit_link . $action, $post->ID ) );
+	if ( 'wp_template' === $post->post_type || 'wp_template_part' === $post->post_type ) {
+		$slug = urlencode( get_stylesheet() . '//' . $post->post_name );
+		$link = admin_url( sprintf( $post_type_object->_edit_link, $post->post_type, $slug ) );
 	} else {
-		$link = '';
+		if ( $post_type_object->_edit_link ) {
+			$link = admin_url( sprintf( $post_type_object->_edit_link . $action, $post->ID ) );
+		} else {
+			$link = '';
+		}
```

Comment:
I don't think an `if/else` block is necessary here. `If` is quite sufficient. But this is nitpicking.
```suggestion
		$link = '';
		if ( $post_type_object->_edit_link ) {
			$link = admin_url( sprintf( $post_type_object->_edit_link . $action, $post->ID ) );
		}
```

Reply:
Updated! Good nit :)